From 65754ad6e78321dda5df413c094067b6df442823 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Fri, 2 Jul 2021 12:36:56 +0600
Subject: [PATCH 28/28] amend check gcc version

---
 kpatch-build/kpatch-build | 2 ++--
 1 file changed, 1 insertions(+), 1 deletions(-)

diff --git a/kpatch-build/kpatch-build b/kpatch-build/kpatch-build
index 6238bc8..1885334 100755
--- a/kpatch-build/kpatch-build
+++ b/kpatch-build/kpatch-build
@@ -229,7 +229,7 @@ gcc_version_check() {
 	echo 'void main(void) {}' > "$c"
 	out="$(gcc -c -pg -ffunction-sections -o "$o" "$c" 2>&1)"
 	gccver="$(gcc_version_from_file "$o")"
-	if [[ -n "$OOT_MODULE" ]]; then
+	if [[ -n "$OOT_MODULE" ]] && [[ "$OOT_MODULE" != yes ]]; then
 		kgccver="$(gcc_version_from_file "$OOT_MODULE")"
 	else
 		kgccver="$(gcc_version_from_file "$VMLINUX")"
-- 
2.23.0

